scripts/internal/workflow-helpers/jira.go

```diff
@@ -203,3 +204,31 @@ func UpdateJiraFixVersion(client *jira.Client, issue *jira.Issue, versionID stri
 	}
 	return nil
 }
+
+func UpdateJiraStatus(client *jira.Client, issue *jira.Issue, statusName string) error {
+	transitions, _, err := client.Issue.GetTransitions(issue.ID)
+	if err != nil {
+		return errs.Wrap(err, "failed to get Jira transitions")
+	}
+
+	var transition *jira.Transition
+	for _, t := range transitions {
+		if t.To.Name == statusName {
+			transition = &t
+			break
+		}
+	}
+	if transition == nil {
+		return errs.New("failed to find a Jira transition that changes the status to %s for issue %s", statusName, issue.Key)
+	}
+
+	response, err := client.Issue.DoTransition(issue.ID, transition.ID)
+	res, err2 := io.ReadAll(response.Body)
```

Comment:
Are you sure `response` is a valid object if `err` is not `nil`? It's probably worth mentioning this in a comment because it looks wrong at first glance.